### PR TITLE
Added tests for details of HttpMessageNotReadableException

### LINKED ISSUE #1: Handle specific reasons of HttpMessageNotReadableException
Something like this

```java
    @ExceptionHandler
    public ResponseEntity<Problem> onHttpMessageNotReadable(final HttpMessageNotReadableException e) throws Throwable {
        final Throwable cause = e.getCause();
        if (cause == null) {
            return onException(e);
        } else if (cause instanceof JsonParseException) {
            return onParseException((JsonParseException) cause);
        } else if (cause instanceof JsonMappingException) {
            return onMappingException((JsonMappingException) cause);
        } else {
             ...
        }
    }
```

### LINKED ISSUE #56: Add an explanation for "throws HttpMediaTypeNotAcceptableException"
Please add in the docs why it throws `HttpMediaTypeNotAcceptableException`, changes came with version `0.8.0`.

### PR DESCRIPTION
**Pull request by whiskeysierra**
Fixes #1
Fixes #56

### CODE REVIEW

**Comment by lukasniemeier-zalando** [comment]
regarding #56 :+1: 
regarding #1: 8 month ago the idea was to expose further fields on the `problem` including structured info about the `location` or `path` of the error (if applicable). If we want to back up from this idea :+1:

File: src/test/java/org/zalando/problem/spring/web/advice/io/MessageNotReadableAdviceTraitTest.java [hunk 1]
```diff
@@ -26,6 +26,32 @@ public void missingRequestBody() throws Exception {
                 .andExpect(jsonPath("$.detail", containsString("request body is missing")));
     }
 
-    // TODO: jackson stuff tests
```
Comment (lukasniemeier-zalando) [inline src/test/java/org/zalando/problem/spring/web/advice/io/MessageNotReadableAdviceTraitTest.java]: other common exceptions:
* `InvalidFormatExceptio`

Comment (lukasniemeier-zalando) [inline src/test/java/org/zalando/problem/spring/web/advice/io/MessageNotReadableAdviceTraitTest.java]: other common exceptions:
* `InvalidFormatExceptio`

**Comment by whiskeysierra** [comment]
I looked into the location/path but it will only contain what's part of the message anyway. Adding a new property would in theory require to *register* a new problem type. Feels like overkill to me, especially since nobody should handle that problem programmatically. It's almost certainly a bug.

File: src/test/java/org/zalando/problem/spring/web/advice/io/MessageNotReadableAdviceTraitTest.java (hunk 1 above)
Comment (whiskeysierra) [inline src/test/java/org/zalando/problem/spring/web/advice/io/MessageNotReadableAdviceTraitTest.java]: the first one is a

> Specialized sub-class of {@link JsonMappingException}

and `Throwables.getRootCause` will return *this* if no cause was found. I think we should be safe here. I'll add two tests though.

Comment (whiskeysierra) [inline src/test/java/org/zalando/problem/spring/web/advice/io/MessageNotReadableAdviceTraitTest.java]: > and Throwables.getRootCause will return

Which I'm not using anymore in the current version... :D

**Comment by lukasniemeier-zalando** [comment]
:+1:

**Comment by kmandeville** [comment]
Were there no changes to help improve this issue? I have the same problem.

**Comment by whiskeysierra** [comment]
@kmandeville We cross-linked a couple of issues here. Can you be more specific?

**Pull request merged**